core/src/main/java/org/mule/runtime/core/internal/lifecycle/phases/LifecycleObjectSorter.java

```diff
@@ -33,9 +33,9 @@
   List<Object> getSortedObjects();
 
   /**
-   * Provides the order of objects as reference for initialise/dispose phases
+   * Provides objects that should be initialized during the initialise/dispose phases
    * 
    * @param lookupObjectsForLifecycle lifecycle object list which is ordered based on the type
    */
-  default void setLifeCycleObjectNameOrderMap(Map<String, Object> lookupObjectsForLifecycle) {};
+  default void setLifeCycleObjectList(Map<String, Object> lookupObjectsForLifecycle) {};
```

Comment:
This is not what I meant. Here you changed the name of the method but it is still receiving a `Map`, so the problem is still there. Also you removed the fact that the order of the components is important from the Javadoc.

Reply:
I reverted the change and changed the input parameter instead!